website/src/roadmap/euphoria.md

```diff
@@ -23,7 +23,74 @@ limitations under the License.
 Easy to use Java 8 DSL for the Beam Java SDK. Provides a high-level abstraction of Beam transformations, which is both easy to read and write. Can be used as a complement to existing Beam pipelines (convertible back and forth). You can have a glimpse of the API at [WordCount example]({{ site.baseurl
 }}/documentation/sdks/java/euphoria/#wordcount-example).
 
-This roadmap is in progress. In the meantime, here are some resources:
-
 - JIRA: [dsl-euphoria](https://issues.apache.org/jira/browse/BEAM-4366?jql=project%20%3D%20BEAM%20AND%20component%20%3D%20dsl-euphoria) / [BEAM-3900](https://issues.apache.org/jira/browse/BEAM-3900)
-- Contact: [David Moravek](mailto:david.moravek@gmail.com)
\ No newline at end of file
+- Contact: [David Moravek](mailto:dmvk@apache.org)
+
+## "Salted" join implementation
+
+Implementation of a join, that can handle large scale join of highly skewed data sets. This implementation breaks
+the large keys into multiple splits, using key distribution approximated by count min sketch data structure. 
+
+## Pipeline sampling
+
+In order to pick the right translation for the operator without user interference, we can leverage knowledge from
+previous pipeline runs. We want to provide a convenient and portable way to gather this knowledge.
+
+## Reusing keyed PCollections
+
+If we have a `PCollection` that is already keyed, we can reuse this and simplify operator builder.
+
+Before:
+
+```java
+final PCollection<KV<String, Integer>> counts = ...;
+
+SumByKey.named("SummarizeCounts")
+  .of(counts)
+  .keyBy(KV:getKey)
+  .valueBy(KV::getValue)
+  .output();
+```
+
+After:
+
+```java
+final PCollection<KV<String, Integer>> counts = ...;
+
+SumByKey.named("SummarizeCounts")
+  .ofKeyed(counts)
+  .valueBy(KV::getValue)
```

Comment:
I suppose this could be also omitted (or made default)?

Reply:
We cannot omit that I'm afraid. That would change the API. And it seems to me that it is useful to have key and value extractors in API. One can skip one `MapElements` operation.

Reply:
I'm not sure I understand. If we have keyed `PCollection`, why do we make default only the `key`, and not the `value` from the `KeyValue`?

Reply:
Sorry, my bad. I misunderstand you comment.

Reply:
I guess this make sense, we'd still need optional `valueBy` method though. 👍